src/servestatic/responders.py

```diff
@@ -315,12 +315,20 @@ def get_path_and_headers(self, request_headers):
 
 
 class Redirect:
-    def __init__(self, location, headers=None):
+    def __init__(self, location, headers=None, preserve_query_string=False):
         headers = list(headers.items()) if headers else []
         headers.append(("Location", quote(location.encode("utf8"))))
         self.response = Response(HTTPStatus.FOUND, headers, None)
+        self.preserve_query_string = preserve_query_string
 
     def get_response(self, method, request_headers):
+        query_string = request_headers.get("QUERY_STRING")
+        if query_string and self.preserve_query_string:
+            headers = list(self.response.headers)
+            name, value = headers[-1]
```

Comment:
This needs a comment for clarification - Specifically why is the query string always at position `headers[-1]`, and is this also true when using `ServeStatic` with ASGI?

Reply:
Made this more generic in 0db086d